### DISCUSSION TITLE
backfilling data via OpenDCS

### DISCUSSION THREAD

**Discussion by CarrJuan** [opendcs/opendcs#863] [General]
Good morning OpenDCS experts.  We did an update recently to our Server Operating System along with our DB and we need to backfill data via OpenDCS for our sites.  I know this is probably pretty trivial for some but was wondering what the proper process is to backfill data.  Any assistance is appreciated.

**Comment by MikeNeilson** [comment]
If it's just GOES data, generally you can set the routing spec time window to cover the range.

If you have a local LRGS and it was also offline, you'll need to create another data source that points to something upstream, like the systems hosted by NOAA.

**Comment by ktarbet** [comment]
Here is an example running the routing spec;  It is likely this will create some files in your ./incoming directory if your using file outputs.

```
rs -d 3 -l CON: -S "now -8 hours" -U "now -7 hours" default
```
where:
-d 3   # means debug logging level 3
-l CON:  # means log to the console
-S "now -8hours"    # start 8 hours back from right now
-U "now -1hours"   # finish 1 hour bck from right now.
default   # the name of your routing spec

**Reply by ktarbet** [reply]
older versions of OpenDCS may not support the CON:  , instead you can give a filename.

**Reply by CarrJuan** [reply]
Do you know if OpenDCS 6.5 supports CON?  How about if you needed to pick a certain day to run back let's say between Friday 12/13/2024 at 6pm and Saturday morning 12/14/2024 at 9am?

**Reply by MikeNeilson** [reply]
it does not, but if linux you can use /dev/stdout as the file name to the same effect.

**Reply by CarrJuan** [reply]
We received this error message when trying to run this command.
[hydromet@ibrcpnlxhyd01 incoming]$ rs -d 3 /dev/stdout
Cannot open log file '/home/hydromet/OPENDCS6/routstat//dev/stdout.log': java.io.FileNotFoundException: /home/hydromet/OPENDCS6/routstat/dev/stdout.log (No such file or directory)

**Reply by MikeNeilson** [reply]
Whelp, unfortunately that means you aren't going to be able to use stdout. Apparently it was sometime between your version and 6.8 were the automatic adding of ".log" was moved from the log file creation.

**Comment by ktarbet** [comment]
Yes, you can pick specific days to run.  A good way would be to copy your routing spec 'default' to another 'juan-backfill', then use the routing spec editor in the Decodes editor to set the dates you want. then you can run the routing spec without dates on the command line and the dates you pick in the editor will be used.

![image](https://github.com/user-attachments/assets/51b66547-147d-4cfc-80bc-e697b319fe1e)

**Comment by CarrJuan** [comment]
Question can you create a data source in OpenDCS in the Database to connect to NOAA DCS with username and password to process missing messages?

**Reply by MikeNeilson** [reply]
Yes, it's an LrgsDataSource. The hostnames are the same as the default provided in the LRGS DDS Receive Configuration.

You'll need to make an account to get the username and password to get the data. Additionally it's required to send a NetList to retrieve the data. Said NetList can contain everything, but it has to exist; you *should* limit to the data you actually want.


Also you could do this on your local LRGS and it will automatically connect to backfill when it discovered GOES data is missing.
